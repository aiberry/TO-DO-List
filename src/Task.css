.singeTask {
    border: solid green;
}

.viewUndone {
    color: black;
    text-decoration: none;
}
  
.viewDone {
    color: gray;
    text-decoration: line-through;
}
  