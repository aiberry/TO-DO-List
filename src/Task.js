import React from 'react';
import PropTypes from 'prop-types';

function Task(props) {
  return (
    <li 
    onClick={props.handler} 
    className={'singeTask ' + (props.status === 'done' ? 'classDone' : 'classUndone')}
    >
      {props.name}
    </li>
  );
}

Task.propTypes = {
  handler: PropTypes.func,
  status: PropTypes.string,
  name: PropTypes.string
};

export default Task;
