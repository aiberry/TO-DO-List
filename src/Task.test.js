import React from 'react';
import Task from './Task.js';
import renderer from "react-test-renderer";
import { initStoreState } from './constants/index.js'

describe("Task component", () => {
  test("snapshot renders", () => {
    const component = renderer
      .create(<Task task={initStoreState[0]} 
                    onClick={()=>console.log('onClick')} />);
    const tree = component.toJSON();
    // instance.handleClick();
    expect(tree).toMatchSnapshot();
  });
});
