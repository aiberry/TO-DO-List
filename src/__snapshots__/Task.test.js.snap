// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component snapshot renders 1`] = `
<li
  className="viewUndone"
  onClick={[Function]}
>
  some2
</li>
`;
